docs/source/troubleshooting.md

```diff
@@ -0,0 +1,85 @@
+  Tools for processing these metrics are scripts `process_logs` and `build_graph_from_csv` bundled with Indy Node.
+  Debug metrics can be used to find some insidios problems like memory leaks and some other hard to detect problems.
+
+## Troubleshooting checklist
+
+### First things to look at in logs
+
+- `incorrect <anything except time>` - if you see in logs messages blaming some PREPREPAREs to have `incorrect state trie`, `incorrect audit root hash` or the like it means that we've got data corruption.
+  If this happened next steps would be:
+  - try to understand how many and which nodes got their data corrupted:
+    - if these messages are seen on minority of nodes, and view change doesn't happen, then it is these nodes have data corrupted
+    - if these messages are seen on majority of nodes, but then view change happens and ordering continues normally, then data corruption happened on primary and non-blaming nodes (after view change they should start blaming new primary for incorrect PREPREPAREs, since view change will reduce situation to previous case)
+  - try to understand what exactly is corrupted and why:
+    - look through `journalctl` logs for recent crashes, especially due to memory or disk space errors. 
+      Sometimes crashes can lead to data corruption.
+    - look through node logs for recent view changes - this is quite a complex process, which sometimes led to data corruption in the past (now this is hopefully fixed with correct implementation of PBFT view change)
+    - if warning was on `incorrect state trie` then most likely only state was corrupted.
+      In this case you can try to stop node, delete state and start it again - node should be able to rebuild correct state from transactions log (ledger).
+      If deleting state doesn't help, and node continues complaining on incorrect state trie, then situation is worse, and probably there is a bug
+
+- `incorrect time` - if you see in logs messages blaming some PREPREPAREs to have `incorrect time`, then there is either one of the following:
+  - either primary node, or node complianing on `incorrect time` has local clock set significantly off, if this is the case clock needs to be adjusted
+  - incoming message queues are filled up faster than processed - which can be indirectly confirmed by increased memory consumption.
+    Possible reasons behind this might be:
+    - previous PREPREPARE messages were also discarded, but by some other reason.
+      If this is the case having PREPREPAREs with incorrect time is just a consequence
+    - there was some performance spike, which caused at first some message delays, followed by an avalanche on message requests.
+      This can be fixed by pool restart (using POOL_RESTART transaction)
+    - there is too much load on the pool (very unlikely situation for current Sovrin deployments, but we've seen this situation during load tests against test pools)
+
+### Useful patterns in logs
+
```

Comment:
I would add `connections changed`

Reply:
Yes, that was planned :)

Reply:
Actually I would prefer to use `disconnected from` and `now connected to` since they allow much more convenient filtering when using grep.